test/Microsoft.ML.Tests/TrainerEstimators/MatrixFactorizationTests.cs

```diff
@@ -442,5 +442,47 @@ public void OneClassMatrixFactorizationInMemoryDataZeroBaseIndex()
             // Negative example (i.e., examples can not be found in dataMatrix) is close to 0.15 (specified by s.C = 0.15 in the trainer).
             CompareNumbersWithTolerance(0.141411, testResults[1].Score, digitsOfPrecision: 5);
         }
+
+        /// <summary>
+        /// This test aims at making sure the training pipeline can always be run in a doc site example at
+        /// https://github.com/dotnet/machinelearning-samples/blob/16acc2f55880808bd34f3465e3eec4571565cb89/samples/csharp/getting-started/MatrixFactorization_MovieRecommendation/MovieRecommendation/Program.cs#L46
+        /// </summary>
+        [ConditionalFact(typeof(Environment), nameof(Environment.Is64BitProcess))]
+        void MatrixFactorizationDocSiteExampleTest()
```

Comment:
Official examples should always run. As currently we don't have a mechanism to test sample projects when building ML.NET, this might be the only solution.

Reply:
Is there a way to ensure the example code, and the unit test stay in sync?

Reply:
Maybe they need to be a submodule of ML.NET.

Reply:
I still don't think we should burden this project to run tests of the other github project.. should just set up the samples repo to run unit tests. If we do this for every sample, this increases our PR build/validation time.